source/solvers/postprocessing_cfd.cc

```diff
@@ -31,6 +31,141 @@
 
 using namespace dealii;
 
+template <int dim, typename VectorType>
+double
+calculate_pressure_drop(const DoFHandler<dim> &        dof_handler,
+                        std::shared_ptr<Mapping<dim>>  mapping,
+                        const MPI_Comm &               mpi_communicator,
+                        std::shared_ptr<FESystem<dim>> fe,
+                        const VectorType &             evaluation_point,
+                        const Quadrature<dim> &        cell_quadrature_formula,
+                        const Quadrature<dim - 1> &    face_quadrature_formula,
+                        const unsigned int             inlet_boundary_id,
+                        const unsigned int             outlet_boundary_id)
+{
+  FEValues<dim>     fe_values(*mapping,
+                          *fe,
+                          cell_quadrature_formula,
+                          update_values | update_quadrature_points |
+                            update_JxW_values | update_gradients |
+                            update_hessians);
+  FEFaceValues<dim> fe_face_values(*fe,
+                                   face_quadrature_formula,
+                                   update_values | update_quadrature_points |
+                                     update_normal_vectors | update_JxW_values);
+
+  const FEValuesExtractors::Scalar pressure(dim);
+
+  const unsigned int n_q_points      = cell_quadrature_formula.size();
+  const unsigned int face_n_q_points = face_quadrature_formula.size();
+
+  std::vector<double> present_pressure_values(n_q_points);
+
+  double pressure_upper_boundary = 0;
+  double upper_surface           = 0;
+  double pressure_lower_boundary = 0;
+  double lower_surface           = 0;
```

Comment:
```suggestion
  double pressure_outlet_boundary = 0;
  double outlet_surface           = 0;
  double pressure_inlet_boundary = 0;
  double inlet_surface           = 0;
```

Better to use the inlet/outlet notation here because upper and lower don't make sense if your pipe is horizontal.